features/dd-sdk-android-trace/src/main/java/com/datadog/trace/common/sampling/SpanSamplingRules.java

```diff
@@ -1,209 +1,263 @@
-        String name,
-        String resource,
-        Map<String, String> tags,
-        double sampleRate,
-        int maxPerSecond) {
-      this.service = service;
-      this.name = name;
-      this.resource = resource;
-      this.tags = tags;
-      this.sampleRate = sampleRate;
-      this.maxPerSecond = maxPerSecond;
+    public SpanSamplingRules(List<Rule> rules) {
+        this.rules = Collections.unmodifiableList(rules);
     }
 
-    /**
-     * Validate and create a {@link Rule} from its {@link JsonRule} representation.
-     *
-     * @param jsonRule The {@link JsonRule} to validate.
-     * @return A {@link Rule} if the {@link JsonRule} is valid, {@code null} otherwise.
-     */
-    public static Rule create(JsonRule jsonRule) {
-      String service = SamplingRule.normalizeGlob(jsonRule.service);
-      String name = SamplingRule.normalizeGlob(jsonRule.name);
-      String resource = SamplingRule.normalizeGlob(jsonRule.resource);
-      Map<String, String> tags = jsonRule.tags;
-      if (tags == null) {
-        tags = Collections.emptyMap();
-      }
-      double sampleRate = 1D;
-      if (jsonRule.sample_rate != null) {
-        try {
-          sampleRate = Double.parseDouble(jsonRule.sample_rate);
-        } catch (NumberFormatException ex) {
-          logRuleError(jsonRule, "sample_rate must be a number between 0.0 and 1.0");
-          return null;
-        }
-        if (sampleRate < 0D || sampleRate > 1D) {
-          logRuleError(jsonRule, "sample_rate must be between 0.0 and 1.0");
-          return null;
-        }
-      }
-      int maxPerSecond = Integer.MAX_VALUE;
-      if (jsonRule.max_per_second != null) {
+    public static SpanSamplingRules deserialize(String json) {
+        SpanSamplingRules result = EMPTY;
         try {
-          double parsedMaxPerSeconds = Double.parseDouble(jsonRule.max_per_second);
-          if (parsedMaxPerSeconds <= 0) {
-            logRuleError(jsonRule, "max_per_second must be greater than 0.0");
-            return null;
-          }
-          maxPerSecond = Math.max((int) parsedMaxPerSeconds, 1);
-        } catch (NumberFormatException ex) {
-          logRuleError(jsonRule, "max_per_second must be a number greater than 0.0");
-          return null;
-        }
-      }
-      return new Rule(service, name, resource, tags, sampleRate, maxPerSecond);
+            result = filterOutNullRules(deserializeRules(new JsonReader(new StringReader(json))));
+        } catch (Throwable ex) {
+            log.error("Couldn't parse Span Sampling Rules from JSON: {}", json, ex);
+        }
+        return result;
     }
 
-    private static void logRuleError(JsonRule rule, String error) {
-      log.error("Skipping invalid Span Sampling Rule: {} - {}", rule, error);
+    public static SpanSamplingRules deserializeFile(String jsonFile) {
+        SpanSamplingRules result = EMPTY;
+        try (com.google.gson.stream.JsonReader reader = new JsonReader(new FileReader(jsonFile))) {
```

Comment:
it is already in the import

```suggestion
        try (JsonReader reader = new JsonReader(new FileReader(jsonFile))) {
```